[eprh] Try to fix tests

poteto opened the pull request:
Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:
---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/facebook/react/pull/32662).
* __->__ #32662
* #32661
* #32615

michaelfaith commented on compiler/packages/eslint-plugin-react-compiler/package.json:
```diff
@@ -26,7 +26,7 @@
     "@types/node": "^20.2.5",
     "babel-jest": "^29.0.3",
-    "eslint": "8.57.0",
+    "eslint-v8": "npm:eslint@^8.57.1",
```
The eslint compiler test failures are due to this change.  It's now using v9, which requires different config input for `RuleTester`.  It would be similar to what the hooks plugin tests set up in the v9 tests.
![image](https://github.com/user-attachments/assets/6e5ccef2-d3e4-4400-b202-3736a689eb05)

poteto committed:
[eprh] Move to compiler directory

Moves the plugin into the compiler directory.

- Remove eslint-plugin-react-hooks from bundles.js
- Remove eslint-plugin-react-hooks from ReactVersions.js
- Remove jest.config.js
- Remove babel.config-react-compiler.js
- Replace babel.config.js with copy from eslint-plugin-react-compiler
- Add tsup.config.ts to eslint-plugin-react-hooks
- Add eslint-plugin-react-hooks to compiler release scripts

poteto committed:
[eprh] Try to fix tests


Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

poteto committed:
[eprh] Remove __EXPERIMENTAL__

`__EXPERIMENTAL__` flag doesn't have any use outside of the React repo. Let's remove these flags for now.
